File: src/test/java/edu/kit/iti/formal/stvs/logic/io/xml/XmlSessionExporterTest.java
package edu.kit.iti.formal.stvs.logic.io.xml;

import static org.junit.Assert.assertEquals;

import edu.kit.iti.formal.stvs.model.StvsRootModel;
import java.io.ByteArrayOutputStream;
import java.io.File;

import org.apache.commons.io.FileUtils;
import org.junit.Before;
import org.junit.Test;

/**
 * @author Benjamin Alt
 */
public class XmlSessionExporterTest {

  private XmlSessionExporter exporter;

  @Before
  public void setUp() {
    exporter = new XmlSessionExporter();
  }

  @Test
  public void exportDefault() throws Exception {
    ByteArrayOutputStream result = exporter.export(new StvsRootModel());
    String resultString = new String(result.toByteArray(), "utf-8");
    String expectedString = FileUtils.readFileToString(new File(this.getClass().getResource
        ("session_empty.xml").toURI()), "utf-8");
    assertEquals(TestUtils.removeWhitespace(expectedString), TestUtils.removeWhitespace
        (resultString));
  }
}

File: stvs/src/test/xml/XmlSessionExporterTest.kt
package edu.kit.iti.formal.stvs.logic.io.xml

import edu.kit.iti.formal.stvs.model.StvsRootModel

import org.apache.commons.io.FileUtils
import org.junit.Assert
import org.junit.Before
import org.junit.jupiter.api.Test
import java.io.*

/**
 * @author Benjamin Alt
 */
class XmlSessionExporterTest {
    private var exporter: XmlSessionExporter? = null

    @Before
    fun setUp() {
        exporter = XmlSessionExporter()
    }

    @Test
    @Throws(Exception::class)
    fun exportDefault() {
        val result: ByteArrayOutputStream = exporter.export(StvsRootModel())
        val resultString = String(result.toByteArray(), charset("utf-8"))
        val expectedString =
            FileUtils.readFileToString(File(this.javaClass.getResource("session_empty.xml").toURI()), "utf-8")
        Assert.assertEquals(TestUtils.removeWhitespace(expectedString), TestUtils.removeWhitespace(resultString))
    }
}